infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/configurator/SshKeysConfigurator.java

```diff
@@ -0,0 +1,222 @@
+import org.eclipse.che.workspace.infrastructure.kubernetes.KubernetesClientFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class mounts existing user SSH Keys into a special Kubernetes Secret on user-s namespace.
+ */
+public class SshKeysConfigurator implements NamespaceConfigurator {
+
+  private static final String SSH_KEYS_WILL_NOT_BE_MOUNTED_MESSAGE =
+      "Ssh keys %s have invalid names and can't be mounted to namespace %s.";
+
+  private static final String SSH_BASE_CONFIG_PATH = "/.ssh/";
+  private static final String SSH_PRIVATE_KEYS = "private";
+  private static final String SSH_PRIVATE_KEYS_PATH = SSH_BASE_CONFIG_PATH + SSH_PRIVATE_KEYS;
+
+  private final SshManager sshManager;
+
+  private final KubernetesClientFactory clientFactory;
+
+  private static final Logger LOG = LoggerFactory.getLogger(SshKeysConfigurator.class);
+
+  public static final Pattern VALID_DOMAIN_PATTERN =
+      Pattern.compile("^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$");
+
+  @Inject
+  public SshKeysConfigurator(SshManager sshManager, KubernetesClientFactory clientFactory) {
+    this.sshManager = sshManager;
+    this.clientFactory = clientFactory;
+  }
+
+  @Override
+  public void configure(NamespaceResolutionContext namespaceResolutionContext, String namespaceName)
+      throws InfrastructureException {
+
+    var client = clientFactory.create();
+    List<SshPairImpl> vcsSshPairs = getVcsSshPairs(namespaceResolutionContext);
+
+    List<String> invalidSshKeyNames =
+        vcsSshPairs
+            .stream()
+            .filter(keyPair -> !isValidSshKeyPair(keyPair))
+            .map(SshPairImpl::getName)
+            .collect(toList());
+
+    if (!invalidSshKeyNames.isEmpty()) {
+      String message =
+          format(
+              SSH_KEYS_WILL_NOT_BE_MOUNTED_MESSAGE, invalidSshKeyNames.toString(), namespaceName);
+      LOG.warn(message);
+      // filter
+      vcsSshPairs = vcsSshPairs.stream().filter(this::isValidSshKeyPair).collect(toList());
+    }
+
+    if (vcsSshPairs.size() == 0) {
+      // nothing to provision
+      return;
+    }
+    doProvisionSshKeys(vcsSshPairs, client, namespaceName);
+  }
+
+  /**
+   * Return list of keys related to the VCS (Version Control Systems), Git, SVN and etc. Usually
+   * managed by user
+   *
+   * @param context NamespaceResolutionContext
+   * @return list of ssh pairs
+   */
+  private List<SshPairImpl> getVcsSshPairs(NamespaceResolutionContext context)
+      throws InfrastructureException {
+    List<SshPairImpl> sshPairs;
+    try {
+      sshPairs = sshManager.getPairs(context.getUserId(), "vcs");
+    } catch (ServerException e) {
+      String message = format("Unable to get SSH Keys. Cause: %s", e.getMessage());
+      LOG.warn(message);
+      throw new InfrastructureException(e);
+    }
+    //    Q: do we need that generation for devworkspaces ?
```

Comment:
I'm not sure here -- what would this be used for?

Reply:
I do not remember the reason why it was added in Che7. Let's remove it for now :)